x-pack/plugins/security/public/views/management/edit_role/components/privileges/kibana/privilege_callout_warning.js

```diff
@@ -3,69 +3,159 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
-import React from 'react';
+import React, { Component, Fragment } from 'react';
 import PropTypes from 'prop-types';
 import {
-  EuiCallOut
+  EuiCallOut,
+  EuiLink,
+  EuiFlyout,
+  EuiFlyoutHeader,
+  EuiFlyoutFooter,
+  EuiFlyoutBody,
+  EuiButton,
+  EuiTitle,
+  EuiSpacer,
 } from '@elastic/eui';
 import { NO_PRIVILEGE_VALUE } from '../../../lib/constants';
+import { PrivilegeSpaceTable } from './privilege_space_table';
+import './privilege_callout.less';
 
-export const PrivilegeCalloutWarning = ({ basePrivilege, isReservedRole }) => {
-  if (basePrivilege === 'all') {
-    if (isReservedRole) {
+export class PrivilegeCalloutWarning extends Component {
+  state = {
+    showImpactedSpaces: false
+  };
+
+  render() {
+    const callout = this.getCallout();
+    const flyout = this.getFlyout();
+
+    return (
+      <Fragment>
+        {callout}
+        {flyout}
+      </Fragment>
+    );
+  }
+
+  getCallout = () => {
+    const {
+      basePrivilege,
+      isReservedRole
+    } = this.props;
+
+    if (basePrivilege === 'all') {
+      if (isReservedRole) {
+        return (
+          <Fragment>
+            <EuiCallOut color="warning" iconType="iInCircle" title={'Cannot customize a reserved role\'s space privileges'}>
+              <p>
+                This role always grants full access to all spaces.
+                To customize privileges for individual spaces, you must create a new role.
+              </p>
+            </EuiCallOut>
+            {this.getShowImpactedSpacesLink()}
```

Comment:
Can this show no matter what the base privilege is? So you can always get a full summary list?

Reply:
And put it in line with the "Add space privilege" button.